tick.go

```diff
@@ -70,36 +73,103 @@ func generateContinuousTicks(r render.Renderer, ra sequence.Range, isVertical bo
 
-	rangeDelta := math.Abs(max - min)
-	tickStep := rangeDelta / float64(intermediateTickCount)
+	formatterName := runtime.FuncForPC(reflect.ValueOf(vf).Pointer()).Name()
+	if strings.HasSuffix(formatterName, "FloatValueFormatter") {
+		intermediateTickCount = mathutil.MinInt(intermediateTickCount, defaultTickCountSanityCheck)
 
-	roundTo := mathutil.RoundTo(rangeDelta) / 10
-	intermediateTickCount = mathutil.MinInt(intermediateTickCount, defaultTickCountSanityCheck)
+		nTicks := niceTicks(min, max, intermediateTickCount)
+		if ra.IsDescending() {
+			// Reverse generated ticks.
+			for i := len(nTicks)/2 - 1; i >= 0; i-- {
+				opp := len(nTicks) - 1 - i
+				nTicks[i], nTicks[opp] = nTicks[opp], nTicks[i]
+			}
+		}
+
+		ticks = []Tick{}
+		for _, t := range nTicks {
+			ticks = append(ticks, Tick{
+				Value: t,
+				Label: vf(t),
+			})
+		}
+	} else {
+		rangeDelta := math.Abs(max - min)
+		tickStep := rangeDelta / float64(intermediateTickCount)
+
+		roundTo := mathutil.RoundTo(rangeDelta) / 10
+		intermediateTickCount = mathutil.MinInt(intermediateTickCount, defaultTickCountSanityCheck)
+
+		for x := 1; x < intermediateTickCount; x++ {
+			var tickValue float64
+			if ra.IsDescending() {
+				tickValue = max - mathutil.RoundUp(tickStep*float64(x), roundTo)
+			} else {
+				tickValue = min + mathutil.RoundUp(tickStep*float64(x), roundTo)
+			}
+			ticks = append(ticks, Tick{
+				Value: tickValue,
+				Label: vf(tickValue),
+			})
+		}
 
-	for x := 1; x < intermediateTickCount; x++ {
-		var tickValue float64
 		if ra.IsDescending() {
-			tickValue = max - mathutil.RoundUp(tickStep*float64(x), roundTo)
+			ticks = append(ticks, Tick{
+				Value: min,
+				Label: vf(min),
+			})
 		} else {
-			tickValue = min + mathutil.RoundUp(tickStep*float64(x), roundTo)
+			ticks = append(ticks, Tick{
+				Value: max,
+				Label: vf(max),
+			})
 		}
-		ticks = append(ticks, Tick{
-			Value: tickValue,
-			Label: vf(tickValue),
-		})
 	}
 
-	if ra.IsDescending() {
-		ticks = append(ticks, Tick{
-			Value: min,
-			Label: vf(min),
-		})
+	return ticks
+}
+
+// niceNum would round tick value to the nearest nice number.
+func niceNum(value float64, round bool) float64 {
+	exponent := math.Floor(math.Log10(value))
+	fraction := value / math.Pow(10, exponent)
+
+	var niceFraction float64
+	if round {
+		if fraction < 1.5 {
+			niceFraction = 1
+		} else if fraction < 3 {
+			niceFraction = 2
+		} else if fraction < 7 {
+			niceFraction = 5
+		} else {
+			niceFraction = 10
+		}
 	} else {
-		ticks = append(ticks, Tick{
-			Value: max,
-			Label: vf(max),
-		})
+		if fraction <= 1 {
+			niceFraction = 1
+		} else if fraction <= 2 {
+			niceFraction = 2
+		} else if fraction <= 5 {
+			niceFraction = 5
+		} else {
+			niceFraction = 10
+		}
 	}
 
-	return ticks
+	return niceFraction * math.Pow(10, exponent)
+}
+
+// nixwTicks generates ticks value with a rounded up values.
+func niceTicks(min, max float64, numTicks int) (tickValues []float64) {
```

Comment:
Would be good to add test cases to check what those return for a few cases and works as expected.